example/heatEquation/src/heatEquation.cpp

```diff
@@ -0,0 +1,210 @@
+    using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
+
+    using DevHost = alpaka::dev::DevCpu;
+    using PltfHost = alpaka::pltf::Pltf< DevHost>;
+
+    // Select specific devices
+    DevAcc const devAcc{alpaka::pltf::getDevByIdx<PltfAcc>(0u)};
+    DevHost const devHost{alpaka::pltf::getDevByIdx<PltfHost>(0u)};
+
+    // Get valid workdiv for the given problem
+    uint32_t elemPerThread = 1;
+    alpaka::vec::Vec<Dim, Idx> const extent{numNodesX};
+    using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Idx>;
+    WorkDiv workdiv{alpaka::workdiv::getValidWorkDiv<Acc>(
+        devAcc,
+        extent,
+        elemPerThread,
+        false,
+        alpaka::workdiv::GridBlockExtentSubDivRestrictions::Unrestricted
+        )};
+
+    // Select queue
+    using QueueProperty = alpaka::queue::Blocking;
+    using QueueAcc = alpaka::queue::Queue<Acc, QueueProperty>;
+    QueueAcc queue{devAcc};
+
+    // Initialize host-buffer
+    using BufHost = alpaka::mem::buf::Buf<DevHost, double, Dim, Idx>;
+    // This buffer holds the calculated values
+    BufHost uNextBufHost{alpaka::mem::buf::alloc<double, Idx>(devHost, extent)};
+    // This buffer will hold the current values (used for the next step)
+    BufHost uCurrBufHost{alpaka::mem::buf::alloc<double, Idx>(devHost, extent)};
+
+    double * const pCurrHost{alpaka::mem::view::getPtrNative(uCurrBufHost)};
+    double * const pNextHost{alpaka::mem::view::getPtrNative(uNextBufHost)};
+
+    // Accelerator buffer
+    using BufAcc = alpaka::mem::buf::Buf<DevAcc, double, Dim, Idx>;
+    BufAcc uNextBufAcc{alpaka::mem::buf::alloc<double, Idx>(devAcc, extent)};
+    BufAcc uCurrBufAcc{alpaka::mem::buf::alloc<double, Idx>(devAcc, extent)};
+
+    double * pCurrAcc{alpaka::mem::view::getPtrNative(uCurrBufAcc)};
+    double * pNextAcc{alpaka::mem::view::getPtrNative(uNextBufAcc)};
+
+    // Apply initial conditions for the test problem
+    for( uint32_t i = 0; i < numNodesX; i++ )
+    {
+        pCurrHost[i] =
+            exactSolution(
+                i * dx,
+                0.0
+            );
+    }
+
+    HeatEquationKernel kernel;
+
+    // Copy host -> device
+    alpaka::mem::view::copy(queue, uCurrBufAcc, uCurrBufHost, extent);
+
+    // EXECUTION
+    for( uint32_t step = 0; step < numTimeSteps; step++ )
+    {
+        // Compute next values
+        alpaka::kernel::exec<Acc>(
+            queue,
+            workdiv,
+            kernel,
+            pCurrAcc,
+            pNextAcc,
+            numNodesX,
+            dx,
+            dt);
+
+        // Swap next to curr (shallow copy)
+        std::swap(pCurrAcc,pNextAcc);
+    }
+
+    // Copy device -> host
+    alpaka::mem::view::copy(queue, uNextBufHost, uNextBufAcc, extent);
+    alpaka::wait::wait(queue);
+
+    // Calculate error
+    double maxError = 0.0;
+    for( uint32_t i = 0; i < numNodesX; i++ )
+    {
+        auto const error = abs( pNextHost[ i ] - exactSolution( i * dx, tMax ) );
```

Comment:
you must use `std::abs` because `abs()` is integer only.
This will than also solve the current CI issue that `error` is an integer.

Reply:
This is true, my sloppy coding for the original version.